lib/cisco_node_utils/pim_group_list.rb

```diff
@@ -0,0 +1,132 @@
+# PIM feature
+# Provides configuration of PIM configuration of rp-addresses with group-lists
+#
+# Smitha Gopalan, November 2015
+#
+# Copyright (c) 2014-2015 Cisco and/or its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#-------------------------------------------------------------------------------
+# CLI: ip pim rp-address <rp-address> group-list <group> (under different VRFs)
+#-------------------------------------------------------------------------------
+
+require_relative 'node_util'
+require 'pp'
+
+module Cisco
+  # node_utils class for pim grouplist
+  class PimGroupList < NodeUtil
+    attr_reader :rp_addr, :vrf, :group
+
+    # Constructor with grouplist and vrf
+    # ----------------------------------
+    def initialize(rp_addr, group, vrf='default', instantiate=true)
+      fail ArgumentError unless vrf.is_a? String
+      fail ArgumentError unless vrf.length > 0
+      @rp_addr = rp_addr
+      @group = group
+      @vrf = vrf
+      if @vrf == 'default'
+        @get_args = @set_args = { state: '', addr: @rp_addr, group: @group }
+      else
+        @get_args = @set_args = { state: '', addr: @rp_addr,
+                                  vrf: @vrf, group: @group }
+      end
+      create if instantiate
+    end
+
+    # Create a hash of [rp-addr,grouplist]=>vrf mappings
+    # --------------------------------------------------
+    def self.group_lists
+      hash_final = {}
+      rp_addrs = config_get('pim_rp_address', 'all_group_lists')
+      return hash_final if rp_addrs.nil?
+      rp_addrs.each do |addr_and_group|
+        # Get the RPs under default VRF
+        addr = addr_and_group[0]
+        group = addr_and_group[1]
+        hash_final[addr_and_group] = {}
+        hash_final[addr_and_group]['default'] = PimGroupList.new(addr, group,
+                                                                 'default',
+                                                                 false)
+      end
+      # Getting all custom vrfs rp_Addrs
+      vrf_ids = config_get('vrf', 'all_vrfs')
+      vrf_ids.delete_if { |vrf_id| vrf_id == 'management' }
+      vrf_ids.each do |vrf|
+        get_args = { rp_addr: @rp_addr, vrf: @vrf, group: @group }
+        get_args[:vrf] = vrf
```

Comment:
Why are you setting vrf twice? Also, why `@rp_addr` and @group`? Don't you want to use the ones in the current loop?
ie:
`       get_args = { rp_addr: addr, vrf: vrf, group: group }`